src/demo/js/toggle-dark.js

```diff
@@ -1,22 +1,54 @@
+const icon = document.querySelector(".darkmode i");
+if (getCookie("darkmode") === null) {
+  if (window.matchMedia("(prefers-color-scheme: dark)").matches == true) {
+    darkmode();
+  } else {
+    lightmode();
+  }
+}
+
 function toggleTheme() {
-  const icon = document.querySelector(".darkmode i");
-  /* dark mode on */
   if (document.body.getAttribute("data-theme") !== "dark") {
-    icon.className = "gg-sun";
-    setCookie("darkmode", "on", 9999);
-    document.body.setAttribute("data-theme", "dark");
-  }
-  /* dark mode off */
-  else {
-    icon.className = "gg-moon";
-    setCookie("darkmode", "off", 9999);
-    document.body.removeAttribute("data-theme");
+    /* dark mode on */
+    darkmode();
+  } else {
+    /* dark mode off */
+    lightmode();
   }
 }
 
+function darkmode() {
+  icon.className = "gg-sun";
+  setCookie("darkmode", "on", 9999);
+  document.body.setAttribute("data-theme", "dark");
+}
+
+function lightmode() {
+  icon.className = "gg-moon";
+  setCookie("darkmode", "off", 9999);
+  document.body.removeAttribute("data-theme");
+}
+
```

Comment:
I would try to avoid making `icon` a global variable. Maybe we could change it like this?

```suggestion
function darkmode() {
  document.querySelector(".darkmode i").className = "gg-sun";
  setCookie("darkmode", "on", 9999);
  document.body.setAttribute("data-theme", "dark");
}

function lightmode() {
  document.querySelector(".darkmode i").className = "gg-moon";
  setCookie("darkmode", "off", 9999);
  document.body.removeAttribute("data-theme");
}

```